api/app/Models/User.php

```diff
@@ -48,4 +48,9 @@ public function isAdmin(): bool
     {
         return in_array('ADMIN', $this->roles);
     }
+    public function setEmailAttribute($value)
+    {
+        $this->attributes['email'] = $value;
+        $this->attributes['sub'] = $this->attributes['email']; // to support authentication
+    }
```

Comment:
Any concerns with moving the email->sub logic to a Laravel model mutator here instead of in the GraphQL layer?

Reply:
I have no concerns with this. @tristan-orourke?

Reply:
Good idea, but doesn't quite work. This would rewrite sub any time we update email, which isn't what we want. We wanted to initialize them with the same value, but want to be able to change them independently.

Reply:
I wouldn't worry much about the sub field though. As of recent integration work with Sign in Canada, it turns out the sub won't match the email, it will need to be a random uuid provided by SiC. Still thinking about how to get it.

Reply:
Ah okay, so we should remove this set email mutator and the copyField directive as well @tristan-orourke ?

Reply:
Remove the email mutator but leave the copyField directive. We still need to ensure the sub field is initiaully populated with _something_. When we rework the sub field, it will deserve its own PR.

Reply:
Alright sounds good, thanks Tristan 👍

Reply:
Hmm, that's a good catch.  You're right, it would update the sub field on email change.

Unfortunately it's not going to be easy to put the `copyField` directive back in.  The reason I took it out is because it breaks on the [createPoolCandidate mutation](https://github.com/GCTC-NTGC/gc-digital-talent/blob/f46297bb02c008f2c308bcbb71ef20fc2322b6a0/api/graphql/schema.graphql#L460).  You want the directive to run if there's a user creation but not an update but I couldn't figure out how to get that to work.  Ideally you would attach the directive only to `CreateUserInput` input instead of to the mutation but I couldn't figure out how to get that to work.

Do you have an idea of how to get that work?  Or should I create an issue and hope that it will get solved with the SiC work?  I noticed the different format sub values, too. Should be interesting. 😊

Reply:
So I flailed about for a few more hours on the directive and got nowhere. 😢  Later I found that there's a creating event you can hook into for a model so I switched to that.  This should fill the sub field only on creation and not when the email changes.  I can't really test that, though, since the schema won't allow email updates.